server/service/apple_mdm.go

```diff
@@ -1342,9 +1354,56 @@ func (svc *Service) GetMDMAppleEnrollmentProfileByToken(ctx context.Context, tok
 		return nil, ctxerr.Wrap(ctx, err)
 	}
 
-	enrollURL, err := apple_mdm.AddEnrollmentRefToFleetURL(appConfig.ServerSettings.ServerURL, ref)
-	if err != nil {
-		return nil, ctxerr.Wrap(ctx, err, "adding reference to fleet URL")
+	enrollURL := appConfig.ServerSettings.ServerURL
+	// check if there is a legacy enroll ref and preserve it for backwards compatibility
+	if ref != "" {
+		idpAcct, err := svc.ds.GetMDMIdPAccountByAccountUUID(
+			// use the primary db as the account might have been just inserted
+			ctxdb.RequirePrimary(ctx, true),
+			ref,
+		)
+		if err != nil {
+			return nil, ctxerr.Wrap(ctx, err, "getting MDM IdP account")
+		}
+		if idpAcct.FleetEnrollRef != "" {
+			level.Debug(svc.logger).Log("msg", "using legacy enroll ref", "enroll_ref", idpAcct.FleetEnrollRef)
+			// we have a legacy enroll ref, add it to the enroll URL
+			enrollURL, err = apple_mdm.AddEnrollmentRefToFleetURL(appConfig.ServerSettings.ServerURL, idpAcct.FleetEnrollRef)
+			if err != nil {
+				return nil, ctxerr.Wrap(ctx, err, "adding reference to fleet URL")
+			}
+		}
+	}
+
+	// TODO: Consider renaming/refactoring this method. We need to do more than just getting the
+	// profile here, we also need to the device with an MDM IdP account
+
+	// if we have device info, we need to parse it and associate the device with an MDM IdP account
+	if deviceinfo != "" {
+		// TODO: Consider implementing just what we need from the parser, instead of needing to
+		// mock up a request that we can pass to the parser
```

Comment:
Totally agree, in fact the repo is MIT-licensed and it looks like we only need this part: https://github.com/korylprince/dep-webview-oidc/blob/2dd846a54fed04c16dd227b8c6c31665b4d0ebd8/header/header.go#L189-L220

It's basically just a helper around some pkcs7 verification and plist unmarshaling AFAICT? Both packages that we already depend on.

Looks like we could extract the relevant function with attribution in a comment?